packages/inventory/src/services/inventory.ts

```diff
@@ -14,33 +14,66 @@ import {
   InventoryItemDTO,
   ReservationItemDTO,
   InventoryLevelDTO,
+  TransactionBaseService,
+  ConfigurableModuleDeclaration,
+  MODULE_RESOURCE_TYPE,
 } from "@medusajs/medusa"
 
 import {
   InventoryItemService,
   ReservationItemService,
   InventoryLevelService,
 } from "./"
+import { EntityManager } from "typeorm"
 
 type InjectedDependencies = {
+  manager: EntityManager
   eventBusService: IEventBusService
 }
 
-export default class InventoryService implements IInventoryService {
+export default class InventoryService
+  extends TransactionBaseService
+  implements IInventoryService
+{
   protected readonly eventBusService_: IEventBusService
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
   protected readonly inventoryItemService_: InventoryItemService
   protected readonly reservationItemService_: ReservationItemService
   protected readonly inventoryLevelService_: InventoryLevelService
 
-  constructor({ eventBusService }: InjectedDependencies) {
+  constructor(
+    { eventBusService, manager }: InjectedDependencies,
+    options?: unknown,
+    moduleDeclaration?: ConfigurableModuleDeclaration
+  ) {
+    super(arguments[0])
+
     this.eventBusService_ = eventBusService
 
-    this.inventoryItemService_ = new InventoryItemService({ eventBusService })
+    if (
+      moduleDeclaration?.resources === MODULE_RESOURCE_TYPE.SHARED &&
```

Comment:
suggestion: `if (moduleDeclaration?.resources !== MODULE_RESOURCE_TYPE.SHARED)` then throw as it is not supported at all for now.